[Gradient Compression] Make GradBucket class public

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#53099 [Gradient Compression] Make GradBucket class public**
* #52981 [Gradient Compression] Refactor tensor grouping in PowerSGD

Publish GradBucket APIs for publishing DDP communication hooks.

s/_GradBucket/GradBucket

Differential Revision: [D26721121](https://our.internmc.facebook.com/intern/diff/D26721121/)

Commit:
[Gradient Compression] Make GradBucket class public

Publish GradBucket APIs for publishing DDP communication hooks.

s/_GradBucket/GradBucket

Differential Revision: [D26721121](https://our.internmc.facebook.com/intern/diff/D26721121/)

[ghstack-poisoned]

rohan-varma reviewed (commented):
Looks good, though could you post a screenshot of how the grad bucket docs are rendered?

SciPioneer commented:
> Looks good, though could you post a screenshot of how the grad bucket docs are rendered?

It's not included in ddp_comm_hooks.rst. Will update the rst file in a separate PR.

Commit:
Update on "[Gradient Compression] Make GradBucket class public"


Publish GradBucket APIs for publishing DDP communication hooks.

s/_GradBucket/GradBucket

Differential Revision: [D26721121](https://our.internmc.facebook.com/intern/diff/D26721121/)

[ghstack-poisoned]

Commit:
Update on "[Gradient Compression] Make GradBucket class public"


Publish GradBucket APIs for publishing DDP communication hooks.

s/_GradBucket/GradBucket

Differential Revision: [D26721121](https://our.internmc.facebook.com/intern/diff/D26721121/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
LGTM, please make sure all tests pass before landing. Would also be great to expose this in the docs in a follow up change as you mentioend.

Commit:
Update on "[Gradient Compression] Make GradBucket class public"


Publish GradBucket APIs for publishing DDP communication hooks.

s/_GradBucket/GradBucket

Differential Revision: [D26721121](https://our.internmc.facebook.com/intern/diff/D26721121/)

[ghstack-poisoned]
